accepted/2022/version-selection.md

```diff
@@ -0,0 +1,449 @@
+
+### Default SDK roll-forward behavior
+
+> Customer value: This feature satisfies the "intuitive version selection" and "Convenient and consistent infra builds"customer scenarios.
+
+By default, SDK version selection does not roll-forward past a feature band. For example, the following `global.json` file requires the use of .NET 7.0.100 and will accept any version through 7.0.199. It will not allow for the use of 7.0.200 (if 7.0.1xx builds are absent).
+
+```json
+{
+  "sdk": {
+    "version": "7.0.100",
+  }
+}
+```
+
+The feature bands have a reason to exist. With each feature band (`.200`, `.300`, ...), there is typically a significant update in a tools component like Roslyn, MSBuild, or NuGet. There may be compatibility breaks. However, if there are significant breaks, that's probably something that needs to resolve. The feature bands are not intended for intentional breaking changes.
+
+This default behavior is overly strict. Starting with .NET 7, the specified version will be treated as the floor version, with no ceiling. The example above would be equivalent to the following:
+
+```json
+{
+  "sdk": {
+    "version": "7.0.100",
+    "rollForward": "latestMajor",    
+  }
+}
+```
+
+The pre-.NET 7 "pinning" behavior becomes opt-in and can be achieved with the following pattern:
+
+```json
+{
+  "sdk": {
+    "version": "7.0.100",
+    "rollForward": "feature",
+  }
+}
+```
+
+This change will only affect `global.json` files with a `version` of `7.0.0` or greater. It will not affect `global.json` files with `6.0.0` (for example) even if .NET 7 SDK is installed on the machine.
+
+The complete set of [roll-forward options are defined in the `global.json` schema](https://docs.microsoft.com/dotnet/core/tools/global-json#rollforward).
+
+### SDK roll-forward gestures
+
+> Customer value: This feature satisfies the "intuitive version selection" and "Convenient and consistent infra builds"customer scenarios.
+
+We will expand the SDK roll-forward gestures to match the similar capabilities for the runtime. These new gestures would enable controlling SDK roll-forward beyond just `global.json`.
+
+It's useful to consider the options that the host offers for runtime roll-forward, via the following ENVs:
+
+- `DOTNET_ROLL_FORWARD` -- a roll-forward policy like `LatestMajor`.
+- `DOTNET_ROLL_FORWARD_TO_PRERELEASE` -- a boolean value on whether roll-forward candidates should include pre-release versions.
+
+The SDK has similar concepts (per an [earlier design](https://github.com/dotnet/designs/blob/main/accepted/2020/global-json-updates.md)). The most obvious approach is to expose matching ENVs as `DOTNET_SDK` to differentiate:
+
+- `DOTNET_SDK_ROLL_FORWARD`
+- `DOTNET_SDK_ROLL_FORWARD_TO_PRERELEASE`
+
+That takes care of the roll-forward ENVs. Let's look at what the CLI offers for the runtime:
+
+- `--roll-forward` -- specifies roll-forward policy
+
+We can follow the same pattern again, by offering a mirror SDK variant:
+
+-- `--sdk-roll-forward`
```

Comment:
We should sync offline with Chet on all these namings as workloads unfortunately has a lot of similarly named concepts (eg. --from-rollback-file, --from-previous-sdk, --sdk-version, etc)

Reply:
Agreed. Chet already raised the same thing.